# 1. Business Problem

In [1]:
#1.1 Business Objective:

#Analyze sales data to understand how phone models and regions are related. 
#Use this information to improve marketing for new phone models.

In [2]:
#1.2 Constraits
#Model Complexity: The model should be able to handle the complexity of the data while being easy to understand and interpret.

# 2.Data Dictionary

In [3]:
#This dataset contains 6 binary discreate l features, representing to improve the sales

# 3.Data Pre-processing

In [6]:
import pandas as pd  # Used for data manipulation and analysis
from mlxtend.frequent_patterns import apriori, association_rules 

In [7]:
file_path = 'myphonedata.csv'  
df = pd.read_csv(file_path)

In [10]:
df = df.astype(bool)
df
# Every cell in the DataFrame has either a True or False value, indicating the presence or absence of a specific color for each row. 
#Therefore, there are no NaN or empty values in the dataset.

red  white  green  yellow  orange   blue
0    True   True   True   False   False  False
1   False   True  False   False    True  False
2   False   True  False   False   False   True
3    True   True  False   False    True  False
4    True  False  False   False   False   True
5   False   True  False   False   False   True
6    True  False  False   False   False   True
7    True   True  False   False   False   True
8   False  False   True   False   False  False
9    True   True  False   False   False   True
10  False  False  False    True   False  False

In [11]:
# Step 3: Apply the Apriori algorithm to find frequent itemsets
# We can set min_support to a reasonable threshold, such as 0.1 (i.e., 10%)
frequent_itemsets = apriori(df, min_support=0.1, use_colnames=True)

In [13]:
# Step 4: Generate association rules from the frequent itemsets
# We use 'confidence' as the metric, with a minimum confidence threshold (e.g., 70%)
rules = association_rules(frequent_itemsets, metric="confidence", min_threshold=0.7)
rules



antecedents consequents  antecedent support  consequent support   support  \
0    (orange)     (white)            0.181818            0.636364  0.181818   

   confidence      lift  leverage  conviction  zhangs_metric  
0         1.0  1.571429  0.066116         inf       0.444444

The confidence value is 1.0, meaning that whenever "orange" appears, "white" also always appears. 

The lift value of 1.571429 suggests that "white" is 1.57 times more likely to appear when "orange" is present than it would appear in the dataset overall. 
A lift greater than 1 shows a positive association between "orange" and "white

In [14]:
# Step 5: Output the results
print("Frequent Itemsets:")
print(frequent_itemsets)



Frequent Itemsets:
    support            itemsets
0  0.545455               (red)
1  0.636364             (white)
2  0.181818             (green)
3  0.181818            (orange)
4  0.545455              (blue)
5  0.363636        (white, red)
6  0.363636         (blue, red)
7  0.181818     (white, orange)
8  0.363636       (white, blue)
9  0.181818  (white, blue, red)


White is a dominant color, often appearing in combination with other colors, including red and blue.
Green and orange are rarely seen in the dataset, either individually or in combination with other colors.
The combination of white and red or blue and red occurs fairly frequently, suggesting that these pairs of colors have a strong association in the dataset.
The appearance of three-color combinations is rare, as only one combination (white, blue, red) appears with a support of 18.18%

In [15]:
print("\nAssociation Rules:")
print(rules[['antecedents', 'consequents', 'support', 'confidence', 'lift']])



Association Rules:
  antecedents consequents   support  confidence      lift
0    (orange)     (white)  0.181818         1.0  1.571429
